beacon_node/network/src/network_beacon_processor/sync_methods.rs

```diff
@@ -44,6 +45,60 @@ struct ChainSegmentFailed {
     peer_action: Option<PeerAction>,
 }
 
+#[derive(Debug)]
+pub enum LookupSyncProcessingResult {
+    FullyImported,
+    ImportedMissingComponents,
+    Error(ErrorCategory),
+    ParentUnknown(Hash256),
+    Ignored,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum ErrorCategory {
+    /// Internal Errors (not caused by peers).
+    ///
+    /// An internal no-retry error is permanent and block processing should not be
+    /// re-attempted.
+    Internal { retry: bool },
+    /// Errors caused by faulty / malicious peers.
+    ///
+    /// No retry errors are deterministic against the block's root. Re-downloading data
+    /// key-ed by block root MUST result in the same no-retry error (i.e. invalid parent,
+    /// invalid state root, etc).
+    ///
+    /// The error also indicates which block component index is malicious if applicable.
+    Malicious { retry: bool, index: usize },
```

Comment:
Does it make sense to call this `Faulty` instead? we don't know if the peer is malicious - i think it might be more consistent with the terminology used in sync, although it doesn't really matter, a faulty peer and malicious peer both send invalid data.